### PR TITLE
KEYCLOAK-19132 SAML adapter config replacement with environment variables

### PR DESCRIPTION
**Pull request by hokuda**
KEYCLOAK-19132 SAML adapter config replacement with environment variables
This PR enables config value replacement with environmet variables in addtion to system properties.

### CODE REVIEW

**Review by sguilhen — DISMISSED** [review]
LGTM

File: common/src/main/java/org/keycloak/common/util/StringPropertyReplacer.java [hunk 1]
```diff
@@ -51,6 +51,8 @@
     private static final int SEEN_DOLLAR = 1;
     private static final int IN_BRACKET = 2;
 
+    private static Properties systemEnvProperties = new SystemEnvProperties();
```
Comment (mabartos) [inline common/src/main/java/org/keycloak/common/util/StringPropertyReplacer.java:54]: Nitpick
```suggestion
    private static final Properties systemEnvProperties = new SystemEnvProperties();
```

**Review by mabartos — COMMENTED** [review]
@hokuda Thanks for your PR! In general, LGTM! Just a few minor things; it's rather a recommendation than request change. WDYT?

**Comment by hokuda** [comment]
@mabartos Thanks for the suggestion! Looks better to me. I applied the changes.

**Review by mabartos — COMMENTED** [review]
@hokuda Thanks! LGTM! Please don't forget to squash those commits.

**Comment by hokuda** [comment]
@mabartos Squashed. Thanks!

**Review by mabartos — DISMISSED** [review]
@hokuda Thanks!
@hmlnarik Could you please take a look at that? Thanks

**Review by sguilhen — APPROVED** [review]

**Review by sguilhen — DISMISSED** [review]

**Comment by sguilhen** [comment]
@hmlnarik changes look good.

**Review by hmlnarik — DISMISSED** [review]
Approving per @sguilhen and @mabartos 's approvals

**Review by hmlnarik — APPROVED** [review]
Approving per original @sguilhen and @mabartos 's approvals

**Pull request merged**